### PR TITLE
[SPARK-36793][K8S] Support write container stdout/stderr to file

### PR DESCRIPTION
**Pull request by warrenzhu25**
### What changes were proposed in this pull request?
Support write container stdout/stderr to file

### Why are the changes needed?
If users want to sidecar logging agent to send stdout/stderr to external log storage,  only way is to change entrypoint.sh, which might break compatibility with community version.

### Does this PR introduce _any_ user-facing change?
Yes. User can enable this feature by spark config.

### How was this patch tested?
Added UT in BasicDriverFeatureStepSuite and BasicExecutorFeatureStepSuite

### CODE REVIEW

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -1120,6 +1120,22 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.logToFile.enabled</code></td>
+  <td><code>false</code></td>
+  <td>
+   Whether to write executor/driver stdout/stderr as log file
+  </td>
+  <td>3.2.0</td>
```
Comment (HyukjinKwon) [inline docs/running-on-kubernetes.md:1129]: it should be 3.3.0

Comment (warrenzhu25) [reply docs/running-on-kubernetes.md]: Updated. Any other comments?

File: resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh [hunk 2]
```diff
@@ -103,4 +103,14 @@ case "$1" in
...
+if [ -z "$K8S_SPARK_LOG_PATH" ]; then
+  exec /usr/bin/tini -s -- "${CMD[@]}"
+else
+  if [ -n "$K8S_SPARK_LOG_PATH" ]; then
+    mkdir -p "$K8S_SPARK_LOG_PATH"
+  fi
+  # TODO: support custom stdout/stderr file name by spark config
+  K8S_SPARK_LOG_STDOUT=$K8S_SPARK_LOG_PATH/stdout
+  K8S_SPARK_LOG_STDERR=$K8S_SPARK_LOG_PATH/stderr
+  exec /usr/bin/tini -s -- "${CMD[@]}" > >(tee $K8S_SPARK_LOG_STDOUT) 2> >(tee $K8S_SPARK_LOG_STDERR >&2)
```
Comment (shanyu) [inline resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh:115]: Would a simple redirect work? e.g.:
1> $K8S_SPARK_LOG_STDOUT 2> $K8S_SPARK_LOG_STDERR

Comment (warrenzhu25) [reply resource-managers/kubernetes/docker/src/main/dockerfiles/spark/entrypoint.sh:115]: Use tee can make stdout/stderr captured by kubectl logs.

**Comment by warrenzhu25** [comment]
@HyukjinKwon Could you help take a look and remove `stale`  label?

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala [hunk 3]
```diff
@@ -82,7 +82,14 @@ private[spark] class BasicDriverFeatureStep(conf: KubernetesDriverConf)
...
           .withValue(env._2)
           .build()
-      }
+      } ++ {
+      if (conf.get(KUBERNETES_LOG_TO_FILE)) {
+        Seq(new EnvVarBuilder()
+          .withName(ENV_SPARK_LOG_PATH)
+          .withValue(conf.get(KUBERNETES_LOG_TO_FILE_PATH))
+          .build())
+      } else None
```
Comment (martin-g) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala:91]: Does this compile ? `None` (Option) is not a collection.
Maybe `Seq.empty` ?!

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala [hunk 4]
```diff
@@ -177,7 +177,14 @@ private[spark] class BasicExecutorFeatureStep(
...
             .build()
         }
-      }
+      } ++ {
+      if (kubernetesConf.get(KUBERNETES_LOG_TO_FILE)) {
+        Seq(new EnvVarBuilder()
+          .withName(ENV_SPARK_LOG_PATH)
+          .withValue(kubernetesConf.get(KUBERNETES_LOG_TO_FILE_PATH))
+          .build())
+      } else None
```
[the remark this answers is not in the document]
Comment (warrenzhu25) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala:186]: I think `None` also worked, one example is https://github.com/warrenzhu25/spark/blob/f9057749f7d7d25d396c03a8041a0a55e97148ab/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala#L156

**Pull request closed without merging**